### PR TITLE
Adding custom categorical plotting of reef polygons to plot_map().

### PR DESCRIPTION
**Pull request by eco-ben**

### CODE REVIEW

File: src/common.jl [hunk 1]
```diff
@@ -103,10 +105,60 @@ function plot_map!(ga::GeoAxis, gdf::DataFrame; geom_col=:geometry, color=nothin
...
+        xlabel="Longitude",
+        ylabel="Latitude",
+        xticklabelpad=15,
+        yticklabelpad=40,
+        xticklabelsize=10,
+        yticklabelsize=10,
+        aspect=AxisAspect(0.75),
+        xgridwidth=0.5,
+        ygridwidth=0.5,
+    )
```
Comment (ConnectedSystems) [inline src/common.jl]: That double assignment isn't necessary I don't think:

```suggestion
    ga = GeoAxis(
        f[1, 1];
        dest="+proj=latlong +datum=WGS84",
        xlabel="Longitude",
        ylabel="Latitude",
        xticklabelpad=15,
        yticklabelpad=40,
        xticklabelsize=10,
        yticklabelsize=10,
        aspect=AxisAspect(0.75),
        xgridwidth=0.5,
        ygridwidth=0.5,
    )
```

File: src/common.jl [hunk 2]
```diff
@@ -103,10 +105,60 @@ function plot_map!(ga::GeoAxis, gdf::DataFrame; geom_col=:geometry, color=nothin
...
+        LE = PolyElement(; color=col)
+        push!(legend_entries, [LE])
+    end
+
+    legend_entries = Dict(zip(unique_names, legend_entries))
+
+    polys = poly!(ga, plottable, color=palette[color_indices])
+
+    Legend(f[2, 1], collect(values(legend_entries)), collect(keys(legend_entries)), nbanks=3,
+        tellheight=true, tellwidth=false, orientation=:horizontal, labelsize=10)
```
Comment (ConnectedSystems) [inline src/common.jl]: Is the `collect` necessary? I would have expected the iterables to work...

**Review by ConnectedSystems — CHANGES_REQUESTED** [review]
Looks fine to me, just two things to address

File: src/common.jl [hunk 3]
```diff
@@ -8,7 +8,8 @@ using
 
 using
     GLMakie,
-    GeoMakie
+    GeoMakie,
+    ColorSchemes
```
Comment (ConnectedSystems) [inline src/common.jl]: Good that you grouped like imports, but Blue style also says to alphabetize them too

**Review by ConnectedSystems — APPROVED** [review]
I'm okay merging this in after addressing one minor comment. Useful as a convenience function, thank you.

**Comment by eco-ben** [comment]
Rebased branch onto main

**Pull request merged**